docs/source/1.0/spec/core/model.rst

```diff
@@ -2695,6 +2698,262 @@ after adding a member to the ``foo`` trait:
         }
 
 
+.. _trait-diff-rules:
+
+Trait diff rules
+----------------
+
+Backward compatibility rules of a trait can be defined in the ``breakingChanges``
+member of a trait definition. This member is a list of diff rules. Smithy
+tooling that performs semantic diff analysis between two versions of the same
+model can use these rules to detect breaking or risky changes.
+
+.. note::
+
+    Not every kind of breaking change can be described using the
+    ``breakingChanges`` property. Such backward compatibility rules SHOULD
+    instead be described through documentation.
+
+.. list-table::
+    :header-rows: 1
+    :widths: 10 20 70
+
+    * - Property
+      - Type
+      - Description
+    * - path
+      - ``string``
+      - A JSON pointer as described in :rfc:`6901` that points to the values
+        to compare from the original model to the updated model. If omitted
+        or if an empty string is provided (``""``), the entire trait is used
+        as the value for comparison. The provided pointer MUST correctly
+        correspond to shapes in the model.
+    * - change
+      - ``string``
+      - **Required**. The type of change. This value can be set to one of the
+        following:
+
+        - ``add``: The trait or value at the current path was added.
+        - ``remove``: The trait or value at the current path was removed.
+        - ``update``: The trait or value at the current path was changed.
+        - ``any``: The trait or value at the current path was added, removed,
+          or changed.
+        - ``presence``: The trait or value at the current path was either
```

Comment:
I'd rather keep it since it's the most common change type. This gives a better DX, but also less importantly is an optimization since you can do two comparisons from a single selection of values.